### ISSUE TITLE
Adding a null object to FlxGroup

### ISSUE THREAD

**Issue by FlixelCommunityBot** [FlixelCommunity/flixel#65]
Issue [#163](https://github.com/AdamAtomic/flixel/issues/163) by: [pixelomatic](http://github.com/pixelomatic)

I accidentally added a FlxGroup to another FlxGroup without creating it with "new". (So it was pointing to null.) I called FlxG.collide on that group. I did not get a null pointer exception but game slowed down to 15fps and I accidentally found the cause.

**Comment by IQAndreas** [comment]
I'm not sure what to do with this one. The fix is easy, but should it throw an error if you try to `add()` a null object, or just ignore it?

**Comment by IQAndreas** [comment]
I also had a thought on how to do it, keeping in mind this message:

> WARNING: If the group has a maxSize that has already been met, the object will NOT be added to the group!

How about, if it is unable to add it to the group (which would also happen if you pass in a `null` object), the function returns `null`? Otherwise, (as it does now) it returns the object passed in.

**Comment by Dovyski** [comment]
If think this approach (return `null` if not added) may cause some headache for developers. The way I would test to check if something was added would be:

``` javascript
if(group.add(obj) == obj) {
}

// or

if(!group.add(obj)) {
}
```

Both tests evaluate to `true` if `obj == null`. That result means the objected was added to the group, but it's not what happened.

I can't think why someone would like to add a `null` entry to a group. I guess it only happens by mistake, so the method should throw an exception.

**Comment by IQAndreas** [comment]
> I guess it only happens by mistake, so the method should throw an exception.

Does Flixel have any sort of error reporting system, do we use `FlxG.log`, or throw a "vanilla" Flash Error?

**Comment by IQAndreas** [comment]
I found a similar case, but where Flixel this time does something about it.

Inside of `FlxObject::followPath()`:

``` as3
if(Path.nodes.length <= 0)
{
    FlxG.log("WARNING: Paths need at least one node in them to be followed.");
    return;
}
```

So, unless anyone has any objections, I'll do the same inside of `FlxGroup`.

**Comment by moly** [comment]
Closed by 39defb4
